docs/source/api/educate.rst

```diff
@@ -0,0 +1,67 @@
+.. module:: pyvrp.educate
+   :synopsis: Education
+
+
+Education
+=========
+
+The :mod:`pyvrp.educate` module contains classes and methods responsible for educating (improving) a newly created offspring solution.
+This happens just after :mod:`pyvrp.crossover` is performed by the :class:`~pyvrp.GeneticAlgorithm.GeneticAlgorithm`. 
```

Comment:
Slightly off-topic: In view of our long-term ambition to possibly include algorithms other than HGS, to what extent should be keep the docs (and code structure) specific to HGS? E.g., `educate` is a HGS-specific term, and it would probably be more future proof to rename this to `local_search`.

This is probably not the place to discuss this in detail, but it's useful to discuss this soon/open an issue about it. What do you think?

Reply:
I've given this some thought, and settled on educate because it'd also cover non-LS type of educators. I'm not entirely sure what we want to do, but we probably do want to facilitate people hybridizing with, say, ALNS (I recall a paper doing HGS + ALNS) or something else.

This also ties into #46, which might have LS derive from a base Educator class to express this intent more clearly. A discussion about this can probably take place there.

Reply:
OK, I'll quote these comments and continue in #46.